xarray/tests/test_dataset.py

```diff
@@ -2410,6 +2414,27 @@ def test_drop_labels_by_keyword(self):
         with pytest.raises(ValueError):
             data.drop(dim="x", x="a")
 
+    def test_drop_labels_by_position(self):
```

Comment:
These are decent. It's even better if we can have one test which can compare the whole `ds` — similar to what you did here: https://github.com/pydata/xarray/pull/4819/files#diff-2db7f6624707083db4aaab1b62eb11352200ec9d3ac1055de84877912226d7b5R2376